cdatransform/extract/pdc.py

```diff
@@ -51,25 +52,35 @@ def get_files_per_sample(self) -> dict:
                 file_metadata = get_file_metadata(f)
                 if aliquots:
                     for aliquot in aliquots:
-                        sample_id = aliquot["sample_id"]
-                        sample_file_pairs.append({"sample_id": sample_id, "file_metadata": file_metadata})
+                        files_per_sample[aliquot["sample_id"]].append(file_metadata)
                         n += 1
             sys.stderr.write(f"Chunk completed. Wrote {n} sample-file pairs in {time.time() - t0}s\n")
         sys.stderr.write(f"Wrote {n} sample-file pairs in {time.time() - t0}s\n")
 
         t1 = time.time()
-        sample_file_pairs.sort(key=lambda x: x["sample_id"])
```

Comment:
I think this new code is equivalent (it appeared to be when I looked at it) but I'm not 100% sure.

Reply:
lol.. killing a fly with an elephant gun with that groupby
defaultdict magic!
Phil, you are the king because of catching this! It's much better and much more readable!